src/lyra/frontend/cfg_generator.py

```diff
@@ -559,13 +600,28 @@ def visit_Call(self, node, types=None, typ=None):
             return Call(pp, name, arguments, typ)
         elif isinstance(func, ast.Attribute):
             name: str = func.attr
-            arguments = [self.visit(func.value, types, typ)]
+            arguments = [self.visit(func.value, types, typ)]        # target of the call
             arguments.extend([self.visit(arg, types, typ) for arg in node.args])
             return Call(pp, name, arguments, typ)
```

Comment:
Note that here bringing 'typ' down to the arguments is not always correct (e.g., when you call split() on a string, 'typ' will be 'List[str]' but you want the arguments to just be of type 'str'). No need for you to do anything about this, just be aware ;) I will try to fix this problem as soon as possible.

Reply:
Maybe you can just give it no typ/None